File: python/problem1503.py
class Solution:
    def getLastMoment(self, n: int, left: List[int], right: List[int]) -> int:
        leftLen, rightLen = len(left), len(right)
        if leftLen == 0 and rightLen == 0:
            return 0
        if leftLen == 0:
            return n - min(right)
        if rightLen == 0:
            return max(left)
        return max(n - min(right), max(left))

File: Java/problem1503.java
class Solution {
    public int getLastMoment(int n, int[] left, int[] right) {
        if (left.length == 0 && right.length == 0) {
            return 0;
        }
        int leftMax = -1, rightMin = n;
        for (int l : left) {
            leftMax = Math.max(leftMax, l);
        }
        for (int r : right) {
            rightMin = Math.min(rightMin, r);
        }
        if (left.length == 0) {
            return n - rightMin;
        }
        if (right.length == 0) {
            return leftMax;
        }
        return Math.max(leftMax, n - rightMin);
    }
}
